gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+        if not (energy_extent[0] <= energy) and (energy < energy_extent[1]):
+            ss_error = "Specified energy {}".format(energy)
+            ss_error += " is outside the boundaries {}.".format(energy_extent)
+            raise ValueError(ss_error)
+
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+        bin_pos = np.searchsorted(energy_edges_high, energy)
+        bin_edges = Quantity([energy_edges_low[bin_pos], energy_edges_high[bin_pos]])
+
+        return bin_pos, bin_edges
+
+    def plot_image(self, energy, ax=None, save=False):
+        """Plot image for the energy bin containing the specified energy.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy of bin to plot the bg model
+        ax : `~matplotlib.axes.Axes`, optional
+            axes of the figure
+        save : `~bool`, optional
+            flag to turn on/off the saving of the figure in a png file
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            figure with image of bin of the bg model for the
+            selected energy value
+        ax : `~matplotlib.axes.Axes`
+            axes of the figure
+        image : `~matplotlib.image.AxesImage`
+            image of the figure, optional
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import LogNorm
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+        energy = energy.flatten() # flatten
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy)
+        if nvalues != 1:
+            print("Expected exactly 1 value for energy, got {}.".format(nvalues))
+            raise IndexError
+        else:
+            energy = Quantity(energy[0])
+            print("Reqested plot only for 1 energy: {}".format(energy))
+
+        extent = self.image_extent
+        energy_bin_centers = self.spectrum_bin_centers
+
+        # find energy bin containing the specified energy
+        energy_bin, energy_bin_edges = self.find_energy_bin(energy)
+        ss_energy_bin_edges = "[{0}, {1}) {2}".format(energy_bin_edges[0].value,
+                                                      energy_bin_edges[1].value,
+                                                      energy_bin_edges.unit)
+        print("Found energy {0} in bin {1} with boundaries {2}.".format(energy,
+                                                                        energy_bin,
+                                                                        ss_energy_bin_edges))
+
+        # get data for the plot
+        ii = energy_bin
+        data = self.background[ii]
+        energy_bin_center = energy_bin_centers[ii]
+
+        # create plot
+        fig = plt.figure()
+        if ax is None:
+            ax = fig.add_subplot(111)
+
+        fig.set_size_inches(8., 8., forward=True)
+
+        image = ax.imshow(data.value,
```

Comment:
Looks like you don't need this `image` object that `ax.imshow` returns?
If that's the case, then no need to store it in an `image` variable, i.e. remove the `image = ` part.